src/Workspaces/SharedUtilitiesAndExtensions/Workspace/Core/LanguageServices/GeneratedCodeRecognition/IGeneratedCodeRecognitionService.cs

```diff
@@ -3,12 +3,17 @@
 // See the LICENSE file in the project root for more information.
 
 using System.Threading;
+using System.Threading.Tasks;
 using Microsoft.CodeAnalysis.Host;
 
 namespace Microsoft.CodeAnalysis.GeneratedCodeRecognition
 {
     internal interface IGeneratedCodeRecognitionService : ILanguageService
     {
+#if !CODE_STYLE
         bool IsGeneratedCode(Document document, CancellationToken cancellationToken);
+#endif
+
+        Task<bool> IsGeneratedCodeAsync(Document document, CancellationToken cancellationToken);
```

Comment:
I'm not really understanding how this works.  This was made shared code, so it can be used by CodeSytle and Workspaces right?  But i thought CodeStyle can't reference Document.  So how does this actually work?

Reply:
The core CodeStyle projects cannot reference Workspace, but CodeStyle.Fixes project (and C#/VB Fixes projects) do reference the Workspaces layer. This API would be accesible from CodeFixes but not from Analyzers in the CodeStyle layer.